src/ui/theme.tsx

```diff
@@ -89,15 +90,23 @@ const { ThemeProvider, useTheme } = createThemeProvider({
     }),
     "palette": {
         ...(() => {
-            switch (THEME_ID) {
-                case "onyxia":
-                    return defaultPalette;
-                case "france":
-                    return francePalette;
-                case "ultraviolet":
-                    return ultravioletPalette;
-                case "verdant":
-                    return verdantPalette;
+            const paletteOverrides = getPaletteOverride();
+            if (paletteOverrides) {
+                return {
+                    ...defaultPalette,
+                    ...paletteOverrides
+                };
```

Comment:
In this instance, a deep merge would be the most suitable approach.

Reply:
I believe the anticipated outcome is for the selected palette to be overwritten. For instance, an administrator may prefer the 'France' theme, yet wish to modify just a single specific color.